catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_1_0/catalog/source/WfsSource.java

```diff
@@ -0,0 +1,1238 @@
+      factory =
+          new SecureCxfClientFactory(
+              wfsUrl,
+              Wfs.class,
+              initProviders(),
+              new MarkableStreamInterceptor(),
+              this.disableCnCheck,
+              false,
+              null,
+              null,
+              username,
+              password);
+    } else {
+      factory =
+          new SecureCxfClientFactory(
+              wfsUrl,
+              Wfs.class,
+              initProviders(),
+              new MarkableStreamInterceptor(),
+              this.disableCnCheck,
+              false);
+    }
+  }
+
+  private List<?> initProviders() {
+    // We need to tell the JAXBElementProvider to marshal the GetFeatureType
+    // class as an element because it is missing the @XmlRootElement Annotation
+    JAXBElementProvider<GetFeatureType> provider = new JAXBElementProvider<>();
+    Map<String, String> jaxbClassMap = new HashMap<>();
+
+    // Ensure a namespace is used when the GetFeature request is generated
+    String expandedName =
+        new QName(Wfs11Constants.WFS_NAMESPACE, Wfs11Constants.GET_FEATURE).toString();
+    jaxbClassMap.put(GetFeatureType.class.getName(), expandedName);
+    provider.setJaxbElementClassMap(jaxbClassMap);
+    provider.setMarshallAsJaxbElement(true);
+
+    featureCollectionReader = new FeatureCollectionMessageBodyReaderWfs11();
+
+    return Arrays.asList(
+        provider,
+        new WfsResponseExceptionMapper(),
+        new XmlSchemaMessageBodyReaderWfs11(),
+        featureCollectionReader);
+  }
+
+  private void setupAvailabilityPoll() {
+    LOGGER.debug(
+        "Setting Availability poll task for {} minute(s) on Source {}", pollInterval, getId());
+    WfsSourceAvailabilityCommand command = new WfsSourceAvailabilityCommand();
+    long interval = TimeUnit.MINUTES.toMillis(pollInterval);
+    if (availabilityPollFuture == null || availabilityPollFuture.isCancelled()) {
+      if (availabilityTask == null) {
+        availabilityTask = new AvailabilityTask(interval, command, getId());
+      } else {
+        availabilityTask.setInterval(interval);
+      }
+      // Run the availability check immediately prior to scheduling it in a thread.
+      // This is necessary to allow the catalog framework to have the correct
+      // availability when the source is bound
+      availabilityTask.run();
+      // Run the availability check every 1 second. The actually call to
+      // the remote server will only occur if the pollInterval has
+      // elapsed.
+      availabilityPollFuture =
+          scheduler.scheduleWithFixedDelay(
+              availabilityTask,
+              AvailabilityTask.NO_DELAY,
+              AvailabilityTask.ONE_SECOND,
+              TimeUnit.SECONDS);
+    }
+  }
+
+  private void availabilityChanged(boolean isAvailable) {
+    if (isAvailable) {
+      LOGGER.info("WFS source {} is available.", getId());
```

Comment:
Since this is polling, might want to put these at `DEBUG` to avoid spamming the logs heavily